### ISSUE TITLE
Streamdeck+ Dial function

### ISSUE THREAD

**Issue by OLDrGoldleader** [bastianh/streamdeck-vjoy-w4rl0ck#2]
HI, Thanks for a great plugin. I can see many uses for it.
Would it be possible to add some functionality to the dials
The Axis option is great, but I would find it very useful if the Dial Press and LCD Press function was configurable separately to the rotate action so for example rotating the dial would change the axis value and pressing the dial in was a configurable vjoy button press and the lcd push was a separate configurable button press.
Also is there any chance you could create an option that allows 4 separate configurable button presses per dial? with the following
1. Dial rotate CCW 
2. Dial Rotate CCW
3. Dial Press
4. LCD Screen Press
so each of those is a different configurable vjoy button press. 

Thanks again for a great plugin.

**Comment by bastianh** [comment]
Yeah ... the dial needs some more options.. I'll definitly update it for more functionality.

The button presses for CW and CCW I'll probably will do in a seperate action since it won't have to deal with an axis anymore and the display would not have a progress bar in that case.

**Comment by bastianh** [comment]
I have implemented 3 and 4 in the current action and it should work in the latest version .. the dial CW and CCW I'll add in a new action.

**Comment by bastianh** [comment]
I have now added a dial action that does button presses on rotate. Currently each press is 50ms with a 50ms break .. The button presses are queued, so turning the dial fast will cause some "lag" until all presses are done. Let me know it it works.. I might make the time configurable if it's to short or long for some games.

**Comment by OLDrGoldleader** [comment]
That was quick, thank you, I will have a look and let you know.

**Comment by OLDrGoldleader** [comment]
It looks like it does exactly what I hoped, so thank you, I will continue to test over the next couple of days.
I do have 2 questions/requests if it is possible and does not cause any major headache. 

1. Is it possible to choose which vJoy device you wish each button/dial to control or does it have to be selected globally for the stream deck device, I have been using this plugin up until now (https://github.com/ashupp/Streamdeck-vJoy) and the vJoy device is selectable on each button you add, is that possible/easy to implement.
2. is a tiny not really important issue but can the axis value slider be removed from the grraphic screen when the action is a button press? no issue if not, I am just curious.

**Comment by bastianh** [comment]
So lets start with the easy question .) 
Yes.. the slider can be removed and I'll fix it with the next release. Actually there are several premade designs for the display but I have not looked into it. I might also make it configurable for the axis dial so it better fits to different background images.

I also started with making the vjoy device selectable for each button, but it got more complicated with the axis and the toggle buttons. I also feared that maybe after doing profiles with hundreds of buttons for different games that I might need to use a different device for some reason and did not want to update all.
So unfortunatly it's not easy to add back or implement it again.
If the problem is not enough buttons, I'm currently also implementing support for the 4 vjoy pov switches, which also could be used as buttons ( with the disadvantage that you can't press two at once I guess ).
The next easier to do solution to support even more buttons and axis would probably be to build/create a second version of the plugin so you could basically have the plugin installed twice, each for a different vjoy device.

**Comment by OLDrGoldleader** [comment]
Thanks for the quick response.

The Issue is not so much lack of buttons, but having multiple stream decks it would be useful if you could have a different vJoy Device being output by different Stream decks without having to install the plugin multiple times.
I understand your comment about the multiple profiles with lots of buttons having to update each button would be frustrating and time consuming, but having a global selection could also possibly cause issues if the vJoy device was changed accidently.
I also use Joystick Gremlin, I have not had any issues with that as I only  trigger vJoy device 1 with Joystick Gremlin and now vJoy Device 2 with Ashupps plugin and 3 with your plugin via the stream deck as I have not gotten around to changing all my pre built profiles using Ashupps plugin to yours, because I have about 20 profiles and they all have at least 32 vjoy buttons in them. and currently you cannot trigger the same vJoy device with different plugins working together on Streamdeck.
Having the Global Selection is not causing me issues, was just a thought.

**Comment by bastianh** [comment]
Yes.. unfortunalty only one application can control a vjoy device, so if plugin A uses a device it can not be controlled from plugin B, even if only other buttons are used.

I'm using the vjoy device from stream-deck actually as an input device for joystick gremlin .. so I can control joystick gremlin mode changes from the stream deck and also add streamdeck buttons that trigger the same buttons as my hotas buttons for example.
